### ISSUE TITLE
nfcapd can't handle netflow UDP packets padded to a specific size

### ISSUE THREAD

**Issue by scottymuse** [phaag/nfdump#73]
While receiving netflow from certain devices, I was getting millions of errors like this:

Process_v9: flowset zero length error.

I found the devices are sending netflow UDP packets padded to a specific length (e.g. even though the packet only contained the netflow header of 20 bytes and one flowset of 60 bytes, the length of the UDP packet was still 1408 bytes long, adding a bunch of 00 bytes at the end of the packet.) It seems nfcapd was ignoring the flowset count field of the packet and attempting to read in the padding bytes, causing the flowset zero length error.

Here is the hack I implemented in netflow_v9.c to get around this issue:

[netflowdiff.txt](https://github.com/phaag/nfdump/files/1143562/netflowdiff.txt)

tl;dr: If it found flowset_length is 0, and then found that flowset_id is 0 as well, just move on. Don't error out.

However this can probably be fixed better.

**Comment by phaag** [comment]
Actually the implementation documents follows the documentatio from CISCO, which says:
```
Padding  |  Padding should be inserted to align the end of the FlowSet on a 32 bit boundary. 
                     Pay attention that the Length field will include those padding bits.
```
nfdump checks the padding in order to verify the correct flowset layout to check for corrupt flow data. So the exporter should also follow these rules.
If you know of a different or more recent document describing padding, let me know.
